app/page.tsx

```diff
@@ -1,61 +1,97 @@
-"use client"
+"use client";
 //import Link from "next/link";
-import React, { useState } from "react";
+import React, { useEffect, useRef, useState } from "react";
 import * as pdfjsLib from "pdfjs-dist/legacy/build/pdf";
 pdfjsLib.GlobalWorkerOptions.workerSrc = `//cdnjs.cloudflare.com/ajax/libs/pdf.js/3.11.174/pdf.worker.min.js`;
 
 import { PDFDocumentProxy } from "pdfjs-dist/types/src/display/api";
-
-
+import RectangleDrawer from "@/component/RectangleDrawer";
+import { usePdfDoc } from "@/app/context/PdfDocContext"; 
 
 export default function Home() {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  const [pdfDoc,setPdfDoc]=useState<PDFDocumentProxy | null>(null);
+  const { pdfDoc, setPdfDoc } = usePdfDoc(); 
+  const [canvas, setCanvas] = useState<HTMLCanvasElement | null>(null);
+  const [overlayCanvas,setOverlayCanvas] = useState<HTMLCanvasElement | null>(null);
+  const canvasRef = useRef<HTMLCanvasElement | null>(null);
+  const overlayCanvasRef = useRef<HTMLCanvasElement | null>(null);
 
-  const handleFileChange=async(event: React.ChangeEvent<HTMLInputElement>)=>{
-    const file=event.target.files?.[0] || null;
+  const handleFileChange = async (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    const file = event.target.files?.[0] || null;
     // used pdfjs lib to preview pdf and not writing any backend api to handle the
-    if(file && file.type==="application/pdf"){
-      const arrayBuffer=await file.arrayBuffer();
-      const loaded=await pdfjsLib.getDocument({
-        data:arrayBuffer,
-        cMapUrl:"/path/to/cmaps/",
-        cMapPacked:true,
+    if (file && file.type === "application/pdf") {
+      const arrayBuffer = await file.arrayBuffer();
+      const loaded = await pdfjsLib.getDocument({
+        data: arrayBuffer,
+        cMapUrl: "/path/to/cmaps/",
+        cMapPacked: true,
       }).promise;
       setPdfDoc(loaded);
-      renderPdf(loaded,1);
-    }else{
+      renderPdf(loaded, 1);
+    } else {
       alert("upload valid pdf");
     }
-  }
-
-  const renderPdf=async(pdfDoc: PDFDocumentProxy,pageNumber: number)=>{
-    //load the page
-    const page = await pdfDoc.getPage(pageNumber);
-    const viewport = page.getViewport({ scale: 1.5 }); // Adjust scale as needed
-    // canvas context for pdf
-    const canvas = document.getElementById("pdfCanvas") as HTMLCanvasElement;
-    const context = canvas.getContext("2d");
-    if(context){
-    canvas.height = viewport.height;
-    canvas.width = viewport.width;
-      // render pdf in canvas context
-    const renderContext = {
-      canvasContext: context,
-      viewport,
-    };
-    await page.render(renderContext).promise;
-  }else{
-    console.error("Falied to get canvas context");
-  }
-  }
+  };
+
+  const renderPdf = async (pdfDoc: PDFDocumentProxy, pageNumber: number) => {
+    if (canvasRef.current && overlayCanvasRef.current) {
```

Comment:
do we need this ?

Reply:
It is an format for loading pdf in pdfJs